prime_numbers.py

```diff
@@ -0,0 +1,31 @@
+def generate_primes(n=100):
+    """
+    This function generates all prime numbers between 1 to n.
+    A prime number is a natural number greater than 1 that has no positive divisors other than 1 and itself.
+    The numbers 0 and 1 are not prime numbers.
+
+    :param n: The upper limit till where to generate prime numbers. Default is 100.
+    :return: A list of prime numbers.
+    """
+    primes = []
+    for possiblePrime in range(2, n + 1):
+        isPrime = True
+        for num in range(2, int(possiblePrime ** 0.5) + 1):
```

Comment:
Instead of checking up to int(possiblePrime ** 0.5) + 1, you can check up to int(possiblePrime / 2) + 1. It would reduce the complexity of the code.